src/Bicep.Core/Emit/ForSyntaxValidatorVisitor.cs

```diff
@@ -247,6 +275,25 @@ private static bool IsPropertyLoop(SyntaxBase? parent, ForSyntax syntax)
             return parent is ObjectPropertySyntax property && ReferenceEquals(property.Value, syntax);
         }
 
+        /// <summary>
+        /// We cannot compile for-expressions when they are inside function calls or operators. This function
+        /// checks if the specified node allows for-expression usage.
+        /// </summary>
+        /// <param name="syntax">The node to check</param>
+        /// <returns></returns>
+        private static bool? IsPropertyLoopUsagePossibleInside(SyntaxBase syntax)
+        {
+            if(syntax is not ExpressionSyntax)
+            {
+                // not an expression but need to distinguish this state from "not allowed" in an expression node
```

Comment:
Worth using an enum rather than a nullable bool? I'm having a hard time keeping track of what the 3 states mean in the above switch statement.

Reply:
Yeah good pt.

Reply:
Done